onnxruntime/core/providers/shared_library/provider_host_api.h

```diff
@@ -26,6 +26,7 @@ struct Provider {
   // Update provider options from key-value string configuration
   virtual void UpdateProviderOptions(void* /*provider options to be configured*/, const ProviderOptions& /*key-value string provider options*/){};
 
+  virtual void Initialize() = 0;
```

Comment:
can we document these? at least thread-safety (or lack thereof) assumptions would be helpful

Reply:
and Shutdown()?

Reply:
I added some documentation, but we don't say much about thread safety in general. There's no way to do it wrong now at least.